cmd/manager/main.go

```diff
@@ -17,157 +17,98 @@ limitations under the License.
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	openebsiov1alpha1 "github.com/openebs/node-disk-manager/api/v1alpha1"
+	blockdevice "github.com/openebs/node-disk-manager/controllers/blockdevice"
+	blockdeviceclaim "github.com/openebs/node-disk-manager/controllers/blockdeviceclaim"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
 )
 
-//ReconciliationInterval defines the triggering interval for reconciliation operation
-const ReconciliationInterval = 5 * time.Second
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 
-func printVersion() {
-	klog.Infof("Go Version: %s", runtime.Version())
-	klog.Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
-	klog.Infof("operator-sdk Version: %v", sdkVersion.Version)
-	klog.Infof("Version Tag: %s", version.GetVersion())
-	klog.Infof("Git Commit: %s", version.GetGitCommit())
+	utilruntime.Must(openebsiov1alpha1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
 }
 
 func main() {
-	// define klog flags
-	klog.InitFlags(nil)
-	flag.Parse()
-
-	// The logger instantiated here can be changed to any logger
-	// implementing the logr.Logger interface. This logger will
-	// be propagated through the whole operator, generating
-	// uniform and structured logs.
-	logf.SetLogger(logf.ZapLogger(false))
-
-	// Init logging
-	ndmlogger.InitLogs()
-	defer ndmlogger.FlushLogs()
-
-	printVersion()
-
-	namespace, err := k8sutil.GetWatchNamespace()
-	if err != nil {
-		klog.Errorf("Failed to get watch namespace: %v", err)
-		os.Exit(1)
-	}
-
-	// Get a config to talk to the apiserver
-	cfg, err := config.GetConfig()
-	if err != nil {
-		klog.Errorf("Failed to get config: %v", err)
-		os.Exit(1)
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	opts := zap.Options{
+		Development: true,
 	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
 
-	// Become the leader before proceeding
-	err = leader.Become(context.TODO(), "node-disk-manager-lock")
-	if err != nil {
-		klog.Errorf("Failed to become leader, Error: %v", err)
-		os.Exit(1)
-	}
-
-	r := ready.NewFileReady()
-	err = r.Set()
-	if err != nil {
-		klog.Errorf("Checking for /tmp/operator-sdk-ready failed: %v", err)
-		os.Exit(1)
-	}
-	defer r.Unset()
-
-	reconInterval := ReconciliationInterval
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
```

Comment:
We had migrated from zap to klog in NDM. Therefore use only klog for logging.

Reply:
This is a new library , not the orignal zap library:  	"sigs.k8s.io/controller-runtime/pkg/log/zap"

Reply:
okay. But the logger is different. We need to resort to using only one type of logger. Else the log formatting will be different from different code bases.